net/http/impl/server.h

```diff
@@ -370,22 +392,31 @@ class HTTPServerConnection {
         }
       }
 
-      // Only support containers of chars and bytes, this does not cover std::string.
+      // The actual implementation of sending HTTP chunk data.
       template <typename T>
-      inline typename std::enable_if<sizeof(typename std::remove_reference<T>::type::value_type) == 1>::type
-      Send(T&& data) {
+      void SendImpl(T&& data) {
         connection_.BlockingWrite(strings::Printf("%X", data.size()));
         connection_.BlockingWrite(kCRLF);
         connection_.BlockingWrite(data);
         connection_.BlockingWrite(kCRLF);
       }
 
+      // Only support STL containers of chars and bytes, this does not yet cover std::string.
+      template <typename T>
+      inline typename std::enable_if<sizeof(typename std::remove_reference<T>::type::value_type) == 1>::type
+      Send(T&& data) {
+        SendImpl(data);
+      }
+
       // Special case to handle std::string.
-      inline void Send(const std::string& data) {
-        connection_.BlockingWrite(strings::Printf("%X", data.length()));
-        connection_.BlockingWrite(kCRLF);
-        connection_.BlockingWrite(data);
-        connection_.BlockingWrite(kCRLF);
+      inline void Send(const std::string& data) { SendImpl(data); }
+
+      // Support objects that can be serialized as JSON-s via Cereal.
+      template <class T>
+      inline typename std::enable_if<
+          (cerealize::is_cerealizeable<typename std::remove_reference<T>::type>::value)>::type
+      Send(T&& object) {
+        SendImpl(cerealize::JSON(object) + kCRLF);
```

Comment:
This JSON payload can be confusing because of `kCRLF` that is also used in the chunked encoding. It's not obvious that `kCRLF` here stands for the "new-line separator" in the "JSON-per-line" format which is a higher-level abstraction ontop of the chunked encoding.